src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -137,798 +34,233 @@ namespace JSONHelpers
-                {
-                    if (data.deviceId == deviceId)
-                    {
-                        DWORD processId = 0;
-                        GetWindowThreadProcessId(window, &processId);
-
-                        auto processIdIt = data.processIdToHandleMap.find(processId);
-
-                        if (processIdIt == std::end(data.processIdToHandleMap))
-                        {
-                            return false;
-                        }
-                        else if (processIdIt->second != window && IsWindow(processIdIt->second))
-                        {
-                            return true;
-                        }
-                    }
-                }
-            }
-        }
-
-        return false;
-    }
-
-    void FancyZonesData::UpdateProcessIdToHandleMap(HWND window, const std::wstring_view& deviceId)
-    {
-        std::scoped_lock lock{ dataLock };
-        auto processPath = get_process_path(window);
-        if (!processPath.empty())
-        {
-            auto history = appZoneHistoryMap.find(processPath);
-            if (history != std::end(appZoneHistoryMap))
-            {
-                auto& perDesktopData = history->second;
-                for (auto& data : perDesktopData)
-                {
-                    if (data.deviceId == deviceId)
-                    {
-                        DWORD processId = 0;
-                        GetWindowThreadProcessId(window, &processId);
-                        data.processIdToHandleMap[processId] = window;
-                        break;
-                    }
-                }
-            }
-        }
-    }
-
-    std::vector<int> FancyZonesData::GetAppLastZoneIndexSet(HWND window, const std::wstring_view& deviceId, const std::wstring_view& zoneSetId) const
+    std::optional<FancyZonesDataTypes::AppZoneHistoryData> ParseSingleAppZoneHistoryItem(const json::JsonObject& json)
     {
-        std::scoped_lock lock{ dataLock };
-        auto processPath = get_process_path(window);
-        if (!processPath.empty())
-        {
-            auto history = appZoneHistoryMap.find(processPath);
-            if (history != std::end(appZoneHistoryMap))
-            {
-                const auto& perDesktopData = history->second;
-                for (const auto& data : perDesktopData)
-                {
-                    if (data.zoneSetUuid == zoneSetId && data.deviceId == deviceId)
-                    {
-                        return data.zoneIndexSet;
-                    }
-                }
-            }
-        }
-
-        return {};
-    }
-
-    bool FancyZonesData::RemoveAppLastZone(HWND window, const std::wstring_view& deviceId, const std::wstring_view& zoneSetId)
-    {
-        std::scoped_lock lock{ dataLock };
-        auto processPath = get_process_path(window);
-        if (!processPath.empty())
+        FancyZonesDataTypes::AppZoneHistoryData data;
+        if (json.HasKey(L"zone-index-set"))
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.